docs/guides/helmreleases.md

```diff
@@ -48,11 +59,73 @@ The `url` can be any HTTP/S Helm repository URL.
     HTTP/S basic and TLS authentication can be configured for private
     Helm repositories. See the [`HelmRepository` CRD docs](../components/source/helmrepositories.md)
     for more details.
+    
+### Git repository
+
+Charts from Git repositories can be released by declaring a
+`GitRepository`, the source-controller will fetch the contents
+of the repository on an interval and expose it as an artifact.
+
+The source-controller can build and expose Helm charts as
+artifacts from the contents of the `GitRepository` artifact
+(more about this later on in the guide).
+
+There are two caveats you should be aware of:
+
+* To make the source-controller produce a new chart artifact,
+  the `version` in the `Chart.yaml` of the chart **must** be
+  bumped.
+* Chart dependencies **must** be committed to Git, as the
+  source-controller does not attempt to download them. This
+  limitation may be removed in a future release.
+  
+An example `GitRepository`:
+
+```yaml
+apiVersion: source.toolkit.fluxcd.io/v1alpha1
+kind: GitRepository
+metadata:
+  name: podinfo
+  namespace: gitops-system
+spec:
+  interval: 1m
+  url: https://github.com/stefanprodan/podinfo
+  ref:
+    branch: master
+  ignore: |
```

Comment:
feat: add `--ignore` flag to `gotk create source git`? (I see a few other struct fields missing too, just noting this as I'm seeing it in the doc).

Speaking of, do we want corresponding `gotk` commands linked from these yaml definitions in general?

Reply:
> feat: add `--ignore` flag to `gotk create source git`? (I see a few other struct fields missing too, just noting this as I'm seeing it in the doc).

Yep, focus has been to get the API stable first before adding every possible option to the CLI.

> Speaking of, do we want corresponding `gotk` commands linked from these yaml definitions in general?

In the future, probably, the [installation guide](https://toolkit.fluxcd.io/guides/installation/) already does this.